### PR TITLE
Add matrix tests that use the new API (unified API)

### PR DESCRIPTION
**Pull request by dkhaldi**
This PR adapts t the matrix tests to the new API (unified API) and move the old API tests to a new folder "Legacy".
This should be merged after the DPC++ changes are merged:  https://github.com/intel/llvm/pull/7413

### CODE REVIEW

**Comment by cperkinsintel** [comment]
The failing test on windows  ( KernelAndProgram/kernel-bundle-merge-options-env.cpp ) is a known failure that was introduced by some other PR. I believe it's being investigated.

**Review by yubingex007-a11y — APPROVED** [review]

**Comment by dkhaldi** [comment]
@intel/llvm-reviewers-runtime, @steffenlarsen can you please help approve/merge?
The failures are unrelated.

**Review by cperkinsintel — APPROVED** [review]
This PR only adds new tests, it doesn't change any existing ones. The CI failures that are seen are unrelated to these tests, therefore this Looks Good To Me.

Note that these new tests all require the `matrix` feature (or a variant) and we don't provide that feature presently, in part because our CI machines don't have any capable machines yet. We should get that feature added in the near future.

**Comment by dkhaldi** [comment]
@intel/llvm-reviewers-runtime, can you please help merge? The failures are unrelated.

File: SYCL/Matrix/joint_matrix_query_default.cpp [hunk 1]
```diff
@@ -7,7 +7,7 @@
 //===----------------------------------------------------------------------===//
 // REQUIRES: matrix
 
-// RUN: %clangxx -fsycl %s -o %t.out
```
Comment (yubingex007-a11y) [inline SYCL/Matrix/joint_matrix_query_default.cpp:10]: sorry, i should have noticed that  this file should be updated

**Pull request merged**